src/lib.rs

```diff
@@ -101,19 +106,65 @@ pub async fn active_allocation_hashes(
         .indexer_allocations()
 }
 
-/// Generate default topics along with given static topics
+/// Generate content topics for all deployments that are syncing on Graph node
+/// filtering for deployments on an index node
+pub async fn syncing_deployment_hashes(
+    graph_node_endpoint: &str,
+    // graphQL filter
+) -> Vec<String> {
+    get_indexing_statuses(graph_node_endpoint.to_string())
+        .await
+        .map_err(|e| -> Vec<String> {
+            error!("Topic generation error: {}", e);
+            [].to_vec()
+        })
+        .unwrap()
+        .iter()
+        .filter(|&status| status.node != "removed")
+        .map(|s| s.subgraph.clone())
+        .collect::<Vec<String>>()
+}
+
+/// Generate a set of unique topics along with given static topics
 pub async fn generate_topics(
+    coverage: CoverageLevel,
     network_subgraph: String,
     indexer_address: String,
+    graph_node_endpoint: String,
     static_topics: &Vec<String>,
 ) -> Vec<String> {
-    let mut topics = active_allocation_hashes(&network_subgraph, indexer_address).await;
-    for topic in static_topics {
-        if !topics.contains(topic) {
-            topics.push(topic.clone());
+    match coverage {
+        CoverageLevel::Minimal => static_topics.to_vec(),
+        CoverageLevel::OnChain => {
+            let mut topics = active_allocation_hashes(&network_subgraph, indexer_address).await;
+            for topic in static_topics {
+                if !topics.contains(topic) {
+                    topics.push(topic.clone());
+                }
+            }
+            topics
+        }
+        CoverageLevel::Comprehensive => {
+            let active_topics: HashSet<String> =
+                active_allocation_hashes(&network_subgraph, indexer_address)
+                    .await
+                    .into_iter()
+                    .collect();
+            let additional_topics: HashSet<String> =
+                syncing_deployment_hashes(&graph_node_endpoint)
+                    .await
+                    .into_iter()
+                    .collect();
+
+            let mut combined_topics: Vec<String> = static_topics.clone();
+            combined_topics.extend(
+                active_topics
+                    .into_iter()
+                    .chain(additional_topics.into_iter()),
+            );
```

Comment:
this is a bit nit-picky
```suggestion
        let active_and_additional_topics = active_topics.union(&additional_topics);
        let combined_topics: Vec<String> = static_topics
            .iter()
            .chain(active_and_additional_topics)
            .cloned()
            .collect();
```